tools/docker-compose/ansible/roles/sources/templates/supervisor.conf.j2

```diff
@@ -0,0 +1,127 @@
+umask = 022
+minfds = 4096
+nodaemon=true
+
+[program:awx-dispatcher]
+command = make dispatcher
+autorestart = true
+stopasgroup=true
+killasgroup=true
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+stderr_logfile=/dev/stderr
+stderr_logfile_maxbytes=0
+
+[program:awx-receiver]
+command = make receiver
+autorestart = true
+stopasgroup=true
+killasgroup=true
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+stderr_logfile=/dev/stderr
+stderr_logfile_maxbytes=0
+
+[program:awx-wsbroadcast]
+command = make wsbroadcast
+autorestart = true
+autorestart = true
+stopasgroup=true
+killasgroup=true
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+stderr_logfile=/dev/stderr
+stderr_logfile_maxbytes=0
+
+[program:awx-uwsgi]
+command = make uwsgi
+autorestart = true
+stopwaitsecs = 1
+stopsignal=KILL
+stopasgroup=true
+killasgroup=true
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+stderr_logfile=/dev/stderr
+stderr_logfile_maxbytes=0
+
+[program:awx-daphne]
+command = make daphne
+autorestart = true
+stopasgroup=true
+killasgroup=true
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+stderr_logfile=/dev/stderr
+stderr_logfile_maxbytes=0
+
+[program:awx-nginx]
+command = make nginx
+autorestart = true
+stopasgroup=true
+killasgroup=true
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+stderr_logfile=/dev/stderr
+stderr_logfile_maxbytes=0
+
+[program:awx-rsyslogd]
+command = rsyslogd -n -i /var/run/awx-rsyslog/rsyslog.pid -f /var/lib/awx/rsyslog/rsyslog.conf
+autorestart = true
+stopasgroup=true
+killasgroup=true
+redirect_stderr=true
+stdout_events_enabled = true
+stderr_events_enabled = true
+
+[program:awx-receptor]
+command = receptor --config /etc/receptor/receptor.conf
+autorestart = true
+stopasgroup=true
+killasgroup=true
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+stderr_logfile=/dev/stderr
+stderr_logfile_maxbytes=0
+
+[group:tower-processes]
+programs=awx-dispatcher,awx-receiver,awx-uwsgi,awx-daphne,awx-nginx,awx-wsbroadcast,awx-rsyslogd
+priority=5
+
+[program:awx-autoreload]
+command = make awx-autoreload
+autostart = true
+autorestart = true
+stopasgroup=true
+killasgroup=true
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+stderr_logfile=/dev/stderr
+stderr_logfile_maxbytes=0
+stdout_events_enabled = true
+stderr_events_enabled = true
+
+[eventlistener:superwatcher]
+command=stop-supervisor
+events=PROCESS_STATE_FATAL
+autorestart = true
+stderr_logfile=/dev/stdout
+stderr_logfile_maxbytes=0
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+
+[unix_http_server]
+file=/var/run/supervisor/supervisor.sock
+
+[supervisorctl]
+serverurl=unix:///var/run/supervisor/supervisor.sock ; use a unix:// URL  for a unix socket
+
+[rpcinterface:supervisor]
+supervisor.rpcinterface_factory = supervisor.rpcinterface:make_main_rpcinterface
+
+[eventlistener:stdout]
+command = supervisor_stdout
+buffer_size = 100
+events = PROCESS_LOG
+result_handler = supervisor_stdout:event_handler
```

Comment:
I like what you're trying to do here, but this concerns me. These new files seem to account for most of the line additions here. This indicates duplication of the files. If you were to move the existing file, I'm not so worried about that. But the supervisor conf, in particular, will absolutely change in the future we do refactoring of the systems, and duplicated files has in the past, and will in the future, cause drift between those files. We already have a hard time keeping the existing supervisor config files in sync. So at the moment, this is the one thing that causes me to raise my hand and say "don't merge yet"

Reply:
Yeah. I didn't love doing this, but felt like doing what it did with the other files was the path of least resistance. I absolutely agree with you, though. That's one reason this is a draft, for sure. I'm open to suggestions, but off the cuff:
1. Just move the files (might break something; don't know what else uses them, etc)
2. I could copy the files over from the Makefile or the playbook (from within the project) and not worry about putting them in the role templates. Not sure if I'd even put them in the role at that point, though. Oh yeah, the other thing about doing it this way is that it's less flexible for moving around (like using the role somewhere else and expecting it to work the same way).

Open to suggestions or ideas.

Reply:
I get where you're coming from. We have multiple supervisor files at these locations

 - `tools/ansible/roles/dockerfile/templates/supervisor.conf.j2` - for the web container, not obvious from the name
 - `tools/ansible/roles/dockerfile/templates/supervisor_task.conf.j2`
 - `tools/docker-compose/supervisor.conf`

The last one is only used by the docker-compose development environment and contains both web & task services, so you could call it a monolithic collection of services. The docker-compose stuff is categorically never used for production, and we can be much more incautious because of that. If you were to move one of the first 2 supervisor configs, that would trigger much more scrutiny, but you're not.

There is a incongruity where we directly map some files, but template others, as shown in the difference between these 2 items in the current docker-compose.

```
      - "../../docker-compose/supervisor.conf:/etc/supervisord.conf"
      - "../../docker-compose/_sources/database.py:/etc/tower/conf.d/database.py"
```

I completely support refactoring this so that both files work in generally the same way. The structure today is an accident of history.

My only nit pick about where to move would be that `templates/supervisor.conf.j2` may not follow _Ansible_ best practices very well, because we don't need to template anything in it (yet). The more standard Ansible role structure would put it in a files subdirectory like `files/supervisor.conf`, and these are generally expected to get copied over by a role.

Before you go too far into this, I must warn that https://github.com/ansible/awx/pull/13423 is a high-priority item that should land in the next few weeks and will have conflicts with your changes. All the more reason to make sure you keep the change to a single commit and do `git mv` for the moved files so that git has the best chance of seamlessly rebasing the change.

Reply:
Excellent explanation. I'm happy to hear that the one `supervisor.conf` is likely moveable. I thought of another way to do it, but now I like aligning the methods so that they work generally the same way, like you said.

- I will move that `supervisor.conf` and double-check for any references.
- I'll also change it so that it's not a template. That was a cheap shot that I knew I'd want to change, and I didn't see the `files` module/practice until afterwards.
- I'll fix failures that CI caught

Thanks for the heads up about #13423 ! I took a look, and the conflicts don't look too bad. I wouldn't be opposed to rebasing on it/waiting for it to land before we land this one.